pkg/operator/ceph/csi/spec.go

```diff
@@ -394,11 +405,99 @@ func StartCSIDrivers(namespace string, clientset kubernetes.Interface, ver *vers
 	return nil
 }
 
-func StopCSIDrivers(namespace string, clientset kubernetes.Interface) error {
-	logger.Warningf("Skipped removing the CSI driver")
+func StopCSIDrivers(namespace string, clientset kubernetes.Interface, ver *version.Info) error {
+
+	deployProvSTS := ver.Major > KubeMinMajor || (ver.Major == KubeMinMajor && ver.Minor < provDeploymentSuppVersion)
```

Comment:
Agreed. If this is incorrect, however, the checks above in object creation are also incorrect. This is the logic that determines whether deployProvSTS is true in both deletion and creation.

Reply:
@travisn, is there a rationale for this check as-is, or have we stumbled on a longstanding bug?

Reply:
The version check is removed now, right? Is there still a question around it?

Reply:
@travisn, The version check is removed in StopCSIDrivers. However, this version check is copied from StartCSIDrivers, where deployProvSTS is evaluated. If it was wrong here, it's been wrong there since at least 10/31/19.